flags/mod.ts

```diff
@@ -31,36 +240,36 @@ export interface ParseOptions {
    *
    * Defaults to `false`.
    */
-  "--"?: boolean;
+  "--"?: DD;
 
   /** An object mapping string names to strings or arrays of string argument
    * names to use as aliases. */
-  alias?: Record<string, string | string[]>;
+  alias?: A;
 
   /** A boolean, string or array of strings to always treat as booleans. If
    * `true` will treat all double hyphenated arguments without equal signs as
    * `boolean` (e.g. affects `--foo`, not `-f` or `--foo=bar`) */
-  boolean?: boolean | string | string[];
+  boolean?: B | ReadonlyArray<Extract<B, string>>;
 
   /** An object mapping string argument names to default values. */
-  default?: Record<string, unknown>;
+  default?: D & Defaults<B, S>;
 
   /** When `true`, populate the result `_` with everything after the first
    * non-option. */
   stopEarly?: boolean;
 
   /** A string or array of strings argument names to always treat as strings. */
-  string?: string | string[];
+  string?: S | ReadonlyArray<Extract<S, string>>;
 
   /** A string or array of strings argument names to always treat as arrays.
    * Collectable options can be used multiple times. All values will be
    * colelcted into one array. If a non-collectable option is used multiple
```

Comment:
There's a typo here: `colelcted` to `collected`?

Reply:
fixed. Thx @dcdunkan 👍